File: tests/fixtures/lua/loaders.lua
local real = require("real")
dofile("scripts/setup.lua")
local compiled = loadfile("scripts/task.lua")

local function parameter(require, dofile)
  require("parameter-shadow.lua")
  dofile("parameter-shadow.lua")
end

do
  local require = function(path)
    return path
  end
  require("block-shadow.lua")
end

require("after-block")

local require = makeLoader()
require("local-shadow.lua")

File: tests/fixtures/ruby/loaders.rb
require "real"
require_relative "support/local"
load "config/setup.rb"
autoload :Worker, "workers/worker.rb"

def shadowed(require)
  require("parameter.rb")
end

class Loader
  def require(feature)
    feature
  end

  def run
    require("method.rb")
  end
end

require "after-class"
